src/lib.rs

```diff
@@ -482,7 +492,7 @@ fn apply_backspace_direct(input: &str) -> String {
 fn readline_direct(
     mut reader: impl BufRead,
     mut writer: impl Write,
-    validator: &Option<impl Validator>,
+    mut validator: Option<&mut impl Validator>,
```

Comment:
not sure why you need the `mut validator`

Reply:
The `match` block requires the input `validator` is mutable. And when we call it, use `readline_direct(io::stdin().lock(), io::stderr(), self.helper.as_mut())`, it is almost same as before
https://github.com/kkawakam/rustyline/blob/77e7d3d772c2db44307a559dfeb0623e44d25165/src/lib.rs#L519-L523

Reply:
Ok